src/scripts/JIRA/etl-408/etl_408.py

```diff
@@ -0,0 +1,172 @@
+            session_token=session_credentials.token)
+    parquet_dataset = parquet.read_table(
+            source=table_source,
+            filesystem=s3_fs,
+            columns=columns)
+    return parquet_dataset.to_pandas()
+
+def get_invalid_records(aws_session, queue_url, iterations, delete_after_sample=False):
+    """
+    Poll an SQS queue for 10 messages at a time for a number of iterations.
+    If `delete_after_sample` is true, messages received from the queue are deleted.
+    Otherwise, since bootstrap sampling is used, you are likely to receive a
+    large number of duplicates.  If the queue is empty, returns the results
+    collected so far.
+
+    Args:
+        aws_session (boto3.session.Session)
+        queue_url (str): The URL of the SQS queue
+        iterations (int): The number of times to poll the SQS queue
+        delete_after_sample: Whether to delete received messages
+            from the queue after they have been received.
+
+    Returns:
+        pandas.DataFrame: A dataframe representation of all messages received
+    """
+    messages = []
+    sqs = aws_session.client("sqs")
+    for i in range(iterations):
+        try:
+            response = sqs.receive_message(
+                    QueueUrl=queue_url,
+                    MaxNumberOfMessages=10
+            )
+            for message in response["Messages"]:
+                message_body = json.loads(message["Body"])
+                messages.append(message_body)
+            if delete_after_sample:
+                sqs.delete_message_batch(
+                        QueueUrl=queue_url,
+                        Entries=[
+                            {"Id": m["MessageId"], "ReceiptHandle": m["ReceiptHandle"]}
+                             for m in response["Messages"]
+                        ]
+                )
+        except Exception as e:
+            print("Encountered the following error while polling the SQS queue. "
+                  "Returning the messages collected thus far.\n")
+            print(str(e))
+            break
+    messages_df = pd.DataFrame(messages)
+    validation_result_df = pd.DataFrame.from_records(
+            messages_df.validation_result.values)
+    invalid_records = pd.concat([messages_df, validation_result_df], axis=1)
+    return invalid_records
+
+def get_missing_records(syn, study):
+    """
+    Get file produced by Larsson which tracks each record
+
+    Args:
+        aws_session (boto3.session.Session)
+        study (str): The Bridge study identifier
+
+    Returns:
+        pandas.DataFrame
+    """
+    # This version from 2023-04-11
+    reference = pd.read_csv(syn.get(MISSING_DATA_REFERENCE).path)
+    study_reference = reference.query(
+            f"studyId == '{study}' and inStudyProject == 1 and inParquet == 0")
+    study_reference = study_reference.dropna(subset=["recordId"])
+    return study_reference
+
+
+def main():
+    aws_session = boto3.session.Session(region_name="us-east-1")
```

Comment:
@philerooski Thanks for putting this script together.  Is it correct to assume that to run this script, you would have to set up the login to the bridge prod account as the default AWS config?

Reply:
Yes, if you want to interact with the production account (or dev account, or any AWS account) you will need to configure your credentials.